File: src/main/java/net/pterodactylus/sone/web/DeleteSonePage.java
/*
 * Sone - DeleteSonePage.java - Copyright © 2010–2016 David Roden
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package net.pterodactylus.sone.web;

import net.pterodactylus.sone.data.Sone;
import net.pterodactylus.sone.web.page.FreenetRequest;
import net.pterodactylus.util.template.Template;
import net.pterodactylus.util.template.TemplateContext;
import net.pterodactylus.util.web.Method;

/**
 * Lets the user delete a Sone. Of course the Sone is not really deleted from
 * Freenet; merely all references to it are removed from the local plugin
 * installation.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
public class DeleteSonePage extends SoneTemplatePage {

	/**
	 * Creates a new page that will delete a Sone.
	 *
	 * @param template
	 *            The template to render
	 * @param webInterface
	 *            The Sone web interface
	 */
	public DeleteSonePage(Template template, WebInterface webInterface) {
		super("deleteSone.html", template, "Page.DeleteSone.Title", webInterface, true);
	}

	//
	// TEMPLATEPAGE METHODS
	//

	/**
	 * {@inheritDoc}
	 */
	@Override
	protected void processTemplate(FreenetRequest request, TemplateContext templateContext) throws RedirectException {
		super.processTemplate(request, templateContext);
		if (request.getMethod() == Method.POST) {
			if (request.getHttpRequest().isPartSet("deleteSone")) {
				Sone currentSone = getCurrentSone(request.getToadletContext());
				webInterface.getCore().deleteSone(currentSone);
			}
			throw new RedirectException("index.html");
		}
	}

}

File: src/main/kotlin/net/pterodactylus/sone/web/DeleteSonePage.kt
package net.pterodactylus.sone.web

import net.pterodactylus.sone.utils.isPOST
import net.pterodactylus.sone.web.page.FreenetRequest
import net.pterodactylus.util.template.Template
import net.pterodactylus.util.template.TemplateContext

/**
 * Lets the user delete a Sone. Of course the Sone is not really deleted from
 * Freenet; merely all references to it are removed from the local plugin
 * installation.
 */
class DeleteSonePage(template: Template, webInterface: WebInterface):
		SoneTemplatePage("deleteSone.html", template, "Page.DeleteSone.Title", webInterface, true) {

	override fun handleRequest(request: FreenetRequest, templateContext: TemplateContext) {
		if (request.isPOST) {
			if (request.httpRequest.isPartSet("deleteSone")) {
				webInterface.core.deleteSone(getCurrentSone(request.toadletContext))
			}
			throw RedirectException("index.html")
		}
	}

}
